### ISSUE TITLE
3D Adversarial Augmentations for Robust Out-of-Domain Predictions

### ISSUE THREAD

**Issue by zhuoxiao-chen** [zhuoxiao-chen/please_read_paper_every_day#53]
### Paper Link
https://link.springer.com/article/10.1007/s11263-023-01914-7

## Motivation
<img width="1523" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/f21b06e0-8133-4194-ae1a-ac0b18f37288">

<img width="600" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/a7b36301-3560-4154-a0b6-32aeb8f68d9c">

## Methodology

<img width="1520" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/d94f3958-b918-47d5-a686-8aeb4302ed7d">

<img width="600" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/5d1c29e9-01fb-4af8-86e2-447ff7abd0d5">

### Adversarial Vector Field

**Discretize the space of a default bounding box, to create a lattice of uniformly spaced 3D vectors within a 3D bounding box.**
- For the car class we constructed each vector ﬁeld within Bo  with w = 1.8 m, h = 1.6 m, l = 4.6 m and a step size of t = 20 cm resulting in 1656 vectors per vector ﬁeld.
- For the person class we used axis-aligned boxes with the dimensions w = 0.54 m, h = 1.7 m, l = 0.66 m and a step size of t = 5 cm resulting in 5036 vectors per vector ﬁeld.

**Adversarial Loss For 3D Object Detection**
<img width="400" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/0553e479-7583-4aa2-b1d0-da9959dfc32e">

**Untargeted Loss Adversarial Loss For 3D Semantic Segmentation**
<img width="400" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/417ca6f8-f5aa-4993-b0dc-bbe6700faabb">

**Targeted Loss Adversarial Loss For 3D Semantic Segmentation**
<img width="400" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/67d72802-c4f7-40f0-8e4c-8d4c946a2be4">

**Training Procedure** 
- Apply the same vectors to all target objects in every scene in the training set
- minimize the adversarial loss across the entire dataset. 
- This process iteratively updates the vectors, resulting in different deformations of the target objects and ultimately leading to different predictions. 
- Once the vectors have been trained, they can be used for data augmentation to improve the performance of a new model.

**Challenges when extending to 3D semantic segmentation** 
- Relied on ground truth 3D bounding boxes to learn and apply vector ﬁelds. However, for 3D semantic segmentation, 3D bounding boxes may not be available.
- It would require a very high amount of vectors when employing a large vector ﬁeld that covers the entire scene.
- Use an off-the-shelf detector to predicted boxes only for the popular car class (smaller classes, such as person, the performance of 3D object detectors is not as reliable). 
- Using axis-aligned boxes around the points belonging to each instance, for classes other than car (using ground truth instance annotations guarantees the correct coverage of all objects).

**Optical Ray Consistency**
<img width="400" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/73a2ff7e-2def-46a6-ab8f-b5e24a6f3a54">

**Regularizing the Deformations**
- To limit the amount by which the points can be perturbed, we restrict the vectors $v_i$  to have a maximum magnitude of $\epsilon$.
- Also ensure that the resulting deformed shape has smooth surfaces by sampling multiple neighboring vectors.

<img width="200" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/592f3c30-00f4-46a0-b00d-0f1da0bc9db9">

This allows for a smoother transition in depth between neighboring points, as vectors with opposite directions would cancel each other out and result in little or no movement of the affected point.

**Relative Rotation**
<img width="1252" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/a9588097-a4c7-48fe-8f4e-20838d837596">

## Experiments 
<img width="1565" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/4364fdbf-b199-47e4-b079-8faf69e4fe7b">
<img width="1409" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/ed038d4c-c641-494c-baab-18be20ec8b9d">
<img width="911" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/455de1a9-bfda-45eb-940f-5eb8d5e60be2">
<img width="830" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/ec927764-eb54-4b91-b01c-ac04cc9cabc8">
<img width="1447" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/8a6816c9-02ab-4e53-b887-3b665c92f41a">
<img width="1564" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/d77c3a70-2fb2-4afe-941c-c999876dce09">
